modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/internal/TcpDiscoveryNodesRing.java

```diff
@@ -393,7 +421,7 @@ public void clear() {
                 nodesMap.put(locNode.id(), locNode);
 
             nodeOrder = 0;
-            maxInternalOrder = 0;
+            maxInternalOrder = locNode.internalOrder();
```

Comment:
.clear() put locNode in nodes. So in old version of .maxInternalOrder() we called nodes.last() and got this the new maxInternalOrder. I just do it immediately here. I think it's better than create two TreeSet with a same values but in difference ordering.